cni/pkg/ambient/net_linux.go

```diff
@@ -237,6 +237,49 @@ func getVethWithDestinationOf(ip string) (*netlink.Veth, error) {
 	return veth, nil
 }
 
+func findVethLinkForPeerIP(peerIP net.IP) (*netlink.Veth, error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list links: %v", err)
+	}
+	var v netlink.Veth
+	for _, l := range links {
+		if l.Type() != v.Type() {
+			continue
+		}
+		veth := l.(*netlink.Veth)
+		peerIndex, err := getPeerIndex(veth)
+		if err != nil {
+			log.Warnf("failed to get peer index for veth interface %s: %v", veth.Name, err)
+		}
+		peerNs, err := getNsNameFromNsID(veth.Attrs().NetNsID)
+		if err != nil {
+			log.Warnf("failed to get network namespace name from namespace id '%d' for veth interface %s: %v", veth.Attrs().NetNsID, veth.Name, err)
+		}
+		var peerVethFound bool
+		if err := netns.WithNetNSPath(fmt.Sprintf("/var/run/netns/%s", filepath.Base(peerNs)), func(netns.NetNS) error {
```

Comment:
I think there was some discussion in the past that `/var/run/netns` is not a valid assumption on all platforms.

That being said, this is a fallback already anyways so doesn't necessarily need to support 100% of cases - just something to consider.

Reply:
In we always mount the host netns to this path inside the CNI container - so this should be fine.

https://github.com/istio/istio/pull/47444 made it so that the host path we mount to that fixed location can be changed, if needed.